File: app/src/test/java/com/mgaetan89/showsrage/model/SearchResultItem_GetIndexerIdTest.java
package com.mgaetan89.showsrage.model;

import com.google.gson.Gson;
import com.mgaetan89.showsrage.network.SickRageApi;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;

import java.util.Arrays;
import java.util.Collection;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(Parameterized.class)
public class SearchResultItem_GetIndexerIdTest {
	@Parameterized.Parameter(1)
	public int indexerId;

	@Parameterized.Parameter(0)
	public SearchResultItem searchResultItem;

	@Test
	public void getIndexId() {
		assertThat(this.searchResultItem.getIndexerId()).isEqualTo(this.indexerId);
	}

	@Parameterized.Parameters
	public static Collection<Object[]> data() {
		Gson gson = SickRageApi.Companion.getGson();

		return Arrays.asList(new Object[][]{
				{gson.fromJson("{}", SearchResultItem.class), 0},
				{gson.fromJson("{indexer: -1, tvdbid: 123, tvrageid: 456}", SearchResultItem.class), 0},
				{gson.fromJson("{indexer: 0, tvdbid: 123, tvrageid: 456}", SearchResultItem.class), 0},
				{gson.fromJson("{indexer: 1, tvdbid: 123, tvrageid: 456}", SearchResultItem.class), 123},
				{gson.fromJson("{indexer: 2, tvdbid: 123, tvrageid: 456}", SearchResultItem.class), 456},
				{gson.fromJson("{indexer: 3, tvdbid: 123, tvrageid: 456}", SearchResultItem.class), 0},
		});
	}
}

File: app/src/test/kotlin/com/mgaetan89/showsrage/model/SearchResultItem_GetIndexerIdTest.kt
package com.mgaetan89.showsrage.model

import com.mgaetan89.showsrage.network.SickRageApi
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized

@RunWith(Parameterized::class)
class SearchResultItem_GetIndexerIdTest(val searchResultItem: SearchResultItem, val indexerId: Int) {
	@Test
	fun getIndexId() {
		assertThat(this.searchResultItem.getIndexerId()).isEqualTo(this.indexerId)
	}

	companion object {
		@JvmStatic
		@Parameterized.Parameters
		fun data(): Collection<Array<Any>> {
			val gson = SickRageApi.gson

			return listOf(
					arrayOf(gson.fromJson("{}", SearchResultItem::class.java), 0),
					arrayOf(gson.fromJson("{indexer: -1, tvdbid: 123, tvrageid: 456, tvmazeid: 789, tmdbid: 987}", SearchResultItem::class.java), 0),
					arrayOf(gson.fromJson("{indexer: 0, tvdbid: 123, tvrageid: 456, tvmazeid: 789, tmdbid: 987}", SearchResultItem::class.java), 0),
					arrayOf(gson.fromJson("{indexer: 1, tvdbid: 123, tvrageid: 456, tvmazeid: 789, tmdbid: 987}", SearchResultItem::class.java), 123),
					arrayOf(gson.fromJson("{indexer: 2, tvdbid: 123, tvrageid: 456, tvmazeid: 789, tmdbid: 987}", SearchResultItem::class.java), 456),
					arrayOf(gson.fromJson("{indexer: 3, tvdbid: 123, tvrageid: 456, tvmazeid: 789, tmdbid: 987}", SearchResultItem::class.java), 789),
					arrayOf(gson.fromJson("{indexer: 4, tvdbid: 123, tvrageid: 456, tvmazeid: 789, tmdbid: 987}", SearchResultItem::class.java), 987),
					arrayOf(gson.fromJson("{indexer: 5, tvdbid: 123, tvrageid: 456, tvmazeid: 789, tmdbid: 987}", SearchResultItem::class.java), 0)
			)
		}
	}
}
